### PR TITLE
Fix #3651: [RTL] High-fi rotation animation for icon in Topic lesson and Hints and Solution

### LINKED ISSUE #3651: [RTL] High-fi Rotation animation need to support RTL
The icon rotation animation needs to be reversed in Topic lesson. (This applies for HintsAndSolutionFragment too). Basically clockwise animation should be replaced with anti-clockwise in RTL and vice-versa.

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation

This PR fixes #3651. This PR fixes expand/collapse icon for RTL.

## Test with new RTL code passes as expected.
Screenshot (57)

## Test without new RTL codes fails as expected.

Screenshot (60)

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by rt4914 — CHANGES_REQUESTED** [review]
@veena14cs Can you please self review this PR? Looks like there as some unwanted changes also the branch is not up-to-date with develop.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
   <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
   <uses-permission android:name="android.permission.INTERNET" />
-  <!-- TODO(#56): Reenable landscape support. -->
```
Comment (anandwana001) [inline app/src/main/AndroidManifest.xml]: why we are removing this?

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -238,4 +240,4 @@
       android:exported="false"
       tools:node="remove" />
   </application>
-</manifest>
+</manifest>
```
Comment (anandwana001) [inline app/src/main/AndroidManifest.xml]: EOF

File: app/src/main/res/layout/activity_view_binding_adapters_test.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    >
+    <ImageView
+      android:id="@+id/test_drop_down_icon"
+      android:layout_width="48dp"
+      android:layout_height="48dp"
+      android:layout_gravity="center_vertical"
+      android:padding="8dp"
+      android:src="@drawable/ic_arrow_drop_down_black_24dp" />
+  </FrameLayout>
+</layout>
```
Comment (anandwana001) [inline app/src/main/res/layout/activity_view_binding_adapters_test.xml]: EOF

File: app/src/sharedTest/java/org/oppia/android/app/databinding/ViewBindingAdaptersTest.kt [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TOLERANCE = 1e-5f
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/databinding/ViewBindingAdaptersTest.kt:77]: add kdoc

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks, @veena14cs 

I haven't reviewed it in detail, looks like few changes. I will check the tests tomorrow.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (veena14cs) [reply app/src/main/AndroidManifest.xml]: Reverted.

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (veena14cs) [reply app/src/main/AndroidManifest.xml]: Done.

File: app/src/main/res/layout/activity_view_binding_adapters_test.xml (hunk 3 above)
Comment (veena14cs) [reply app/src/main/res/layout/activity_view_binding_adapters_test.xml]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/databinding/ViewBindingAdaptersTest.kt (hunk 4 above)
Comment (veena14cs) [reply app/src/sharedTest/java/org/oppia/android/app/databinding/ViewBindingAdaptersTest.kt:77]: Done.

**Comment by veena14cs** [comment]
> Thanks, @veena14cs
> 
> I haven't reviewed it in detail, looks like few changes. I will check the tests tomorrow.

Thanks @anandwana001 I have addressed your comments PTAL.

**Comment by veena14cs** [comment]
> @veena14cs Can you please self review this PR? Looks like there as some unwanted changes also the branch is not up-to-date with develop.

Fixed unwanted changes PTAL.

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java [hunk 5]
```diff
@@ -45,23 +44,25 @@ public static void setLayoutWidth(@NonNull View view, float width) {
...
+      }
+    });
+    valueAnimator.start();
+  }
+
+  private static boolean isRtlLayout(View view) {
+    return ViewCompat.getLayoutDirection(view) == ViewCompat.LAYOUT_DIRECTION_RTL;
+  }
 }
+
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: Remove this extra line.

File: app/src/main/res/layout/activity_view_binding_adapters_test.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto">
+  <FrameLayout
+    android:id="@+id/expand_list_icon"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="8dp"
+    android:layout_marginEnd="8dp"
+    android:layout_marginBottom="8dp"
+    >
```
Comment (rt4914) [inline app/src/main/res/layout/activity_view_binding_adapters_test.xml]: Press `Control + Alt + L` - reformatting shortcut to fix the formatting.

**Review by rt4914 — APPROVED** [review]
Suggested nit changes. Otherwise LGTM

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java (hunk 5 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: Done.

File: app/src/main/res/layout/activity_view_binding_adapters_test.xml (hunk 6 above)
Comment (veena14cs) [reply app/src/main/res/layout/activity_view_binding_adapters_test.xml]: Done.

**Review by anandwana001 — APPROVED** [review]
LGTM

**Comment by BenHenning** [comment]
Hi. Will need to look at this tomorrow.

**Review by BenHenning — APPROVED** [review]
Thanks @veena14cs. I'm glad we were able to effectively test this. PR LGTM!

**Pull request merged**